google/cloud/sql/connector/connector.py

```diff
@@ -17,6 +17,54 @@
 import googleapiclient
 
 
+def get_metadata(service, proj_name, inst_name):
+    """
+    A helper function that requests metadata from the Cloud SQL Instance
+    and returns a dictionary containing the IP addresses and certificate
+    authority of the Cloud SQL Instance.
+
+    Args:
+        service (googleapiclient.discovery.Resource): A service object created
+          from the Google Python API client library. Must be using the SQL
+          Admin API. For more info check out
+          https://github.com/googleapis/google-api-python-client.
+        proj_name (str): A string representing the name of the project.
+        inst_name(str): A string representing the name of the instance.
+            Usually found in environment variable 'CLOUD_SQL_INSTANCE_NAME.'
```

Comment:
if this is a common env var would it make sense for us to identify this and make this an optional param?

Similarly for project. The `ClientWithProject` class in `google-cloud-python/core` makes project optional and will identify the project automagically.

Reply:
I missed this before - but generally they'll specify the instance name in the connect method (we probably don't want to advertise they should set it via this env var)